### PR TITLE
S3 website routing rules

### PR DESCRIPTION
**Pull request by dminnear**
@Bowbaq This should be good to go.

### CODE REVIEW

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 1]
```diff
@@ -684,6 +700,15 @@ func resourceAwsS3BucketWebsitePut(s3conn *s3.S3, d *schema.ResourceData, websit
 		}
 	}
 
+	if routingRules != "" {
+		var unmarshaledRules []*s3.RoutingRule
+		if err := json.Unmarshal([]byte(routingRules), &unmarshaledRules); err != nil {
+			return err
+		} else {
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket.go]: Don't use `else` here. Either you return, or just keep going with the normal flow of code

```golang
if err := json.Unmarshal([]byte(routingRules), &unmarshaledRules); err != nil {
  return err
} 

websiteConfiguration.RoutingRules = unmarshaledRules
```

File: builtin/providers/aws/resource_aws_s3_bucket.go [hunk 2]
```diff
@@ -854,6 +920,19 @@ func normalizeJson(jsonString interface{}) string {
 	return string(b[:])
 }
 
+func normalizeJsonArray(jsonString interface{}) string {
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket.go]: If you just change line 914 to `var j interface{}` you can just reuse `normalizeJson`. https://play.golang.org/p/IecCCed0OE

File: builtin/providers/aws/resource_aws_s3_bucket_test.go [hunk 3]
```diff
@@ -452,6 +498,31 @@ func testAccCheckAWSS3BucketWebsite(n string, indexDoc string, errorDoc string,
...
+		conn := testAccProvider.Meta().(*AWSClient).s3conn
+
+		out, err := conn.GetBucketWebsite(&s3.GetBucketWebsiteInput{
+			Bucket: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if routingRules == nil {
+				return nil
+			} else {
```
Comment (Bowbaq) [inline builtin/providers/aws/resource_aws_s3_bucket_test.go]: same thing with the `else` clause here

**Comment by Bowbaq** [comment]
few minor nitpicks but generally LGTM

**Comment by dminnear** [comment]
Already in TF proper. Closing.

**Pull request closed without merging**